### DISCUSSION TITLE
Get highlight group for places where the cursor cannot get to

### DISCUSSION THREAD

**Discussion by Nargonath** [neovim/neovim#27257] [Q&A]
I help maintain a [nvim colorscheme](https://github.com/spaceduck-theme/nvim) and sometimes I have instances where something doesn't look right. I want to fix the color but I don't how to find the name of the specific highlight group. In Neovim 0.9, I used `Inspect` but it works only where the cursor can be placed. For example let's say I want to know the highlight group used for the line separator between two buffers. Is there a way for me to put my mouse cursor on it and invoke a command that'd the same as `Inspect`? For now I used Telescope highlight picker and filter through trying to find what I want and change colors to see if it apply to where I want. This is rather slow and cumbersome so I'm trying to find a better way.

Thank you for your help.

**Comment by wookayin** [comment]
:help vim.inspect_pos()

```
vim.inspect_pos({bufnr}, {row}, {col}, {filter})           *vim.inspect_pos()*
    Get all the items at a given buffer position.
```

Although you have to figure out the bufnr and row/col; this would be only useful when the cursor can't jump to a (floating) window that might be gone as soon as the cursoris moved.

However, this can't be used on *non-buffer* highlights such as the line separator as in your case. Although it's quite non-trivial, one option is to use the RPC UI API (:help ui-events, :help ui-linegrid) --- it can access to all the TUI drawing information about the grid to be rendered. See also `test/functional/ui/screen.lua` for how this is done in neovim's functional tests.

**Reply by Nargonath** [reply]
Ok thank you @wookayin for your answer. I'll try that out and report back.
